GameOfLifeApp/GameLogic.cs

```diff
@@ -0,0 +1,153 @@
+        /// <returns>Returns number of alive neighbours.</returns>
+        private int NeighboursCount(int currentRow, int currentColumn)
+        {
+            int count = 0;
+            for (int row = currentRow - 1; row <= currentRow + 1; row++)
+            {
+                for (int column = currentColumn - 1; column <= currentColumn + 1; column++)
+                {
+                    int actualRow;
+                    int actualColumn;
+                    if (row < 0)
+                    {
+                        actualRow = CurrentField.GetLength(0) - 1;
+                    }
+                    else if (row == CurrentField.GetLength(0))
+                    {
+                        actualRow = 0;
+                    }
+                    else
+                    {
+                        actualRow = row;
+                    }
+                    if (column < 0)
+                    {
+                        actualColumn = CurrentField.GetLength(1) - 1;
+                    }
+                    else if (column == CurrentField.GetLength(1))
+                    {
+                        actualColumn = 0;
+                    }
+                    else
+                    {
+                        actualColumn = column;
+                    }
+                    count += CurrentField[actualRow, actualColumn] ? 1 : 0;
+                }
+            }
+            count -= CurrentField[currentRow, currentColumn] ? 1 : 0;
+
+            return count;
+        }
+
+        /// <summary>
+        /// Gets new boolean values according to statements.
+        /// </summary>
+        /// <param name="neighboursCount">Number of alive cells around a given cell.</param>
+        /// <param name="isCurrentAlive">Given cell is alive. Value of isCurrentAlive: true.</param>
+        /// <returns>Returns the next value for a given cell.</returns>
+        private bool GetCellOffsprings(int neighboursCount, bool isCurrentAlive)
+        {
+            if (!isCurrentAlive && neighboursCount == 3)
+            {
+                return true;
+            }
+            else if (isCurrentAlive && neighboursCount == 2 || neighboursCount == 3)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// This method generates a new field and transfers data to a generated field.       
+        /// </summary>
+        public void GetNextGeneration()
+        {
+            bool[,] generatedField = new bool[CountOfRows, CountOfColumns];
+
+            for (int row = 0; row < CountOfRows; row++)
+            {
+                for (int column = 0; column < CountOfColumns; column++)
+                {
+                    int count = NeighboursCount(row, column);
+                    bool value = GetCellOffsprings(count, CurrentField[row, column]);
+
+                    generatedField[row, column] = value;
```

Comment:
if the value variable is only used in this one place then you can just write it like this 
`generatedField[row, column] = GetCellOffsprings(count, CurrentField[row, column]);`
no need to just create a variable and use it where you can just call the method itself.

Reply:
Done.